src/main/java/org/threadly/litesockets/Client.java

```diff
@@ -3,557 +3,358 @@
  * 
- * The client object can not functions with out being in a SocketExecuter.
+ * The client object can not function with out being in a SocketExecuter.  Writes can be added before its put
+ * in the executer, but it will not write to the socket until added to the executer.
  * 
  * @author lwahlmeier
  *
  */
-public abstract class Client {
-  public static final int DEFAULT_MAX_BUFFER_SIZE = 64*1024;
-  public static final int MIN_READ= 4*1024;
-  
-  private final AtomicInteger readBufferSize = new AtomicInteger(0);
-  private final AtomicInteger writeBufferSize = new AtomicInteger(0);
-  
-  protected ClientByteStats stats = new ClientByteStats();
-  
-  protected int maxBufferSize = DEFAULT_MAX_BUFFER_SIZE;
-  protected int minAllowedReadBuffer = MIN_READ;
-  
-  private final MergedByteBuffers readBuffers = new MergedByteBuffers();
-  private final MergedByteBuffers writeBuffers = new MergedByteBuffers();
-  private ByteBuffer currentWriteBuffer;
-  private ByteBuffer readByteBuffer = ByteBuffer.allocate(maxBufferSize*2);
-  
-  protected volatile Closer closer;
-  protected volatile Reader reader;
-  protected volatile SubmitterExecutorInterface sei;
-  protected volatile SocketExecuterBase ce;
-  protected AtomicBoolean closed = new AtomicBoolean(false);
+public interface Client {
   
   /**
-   * Returns the SimpleByteStats object for this client.
+   * Returns true if this client can have reads added to it or false if its read buffers are full.
    * 
-   * @return the byte stats for this client.
-   */
-  public SimpleByteStats getStats() {
-    return stats;
-  }
-  
-  /**
-   * Returns true if this client can read or false if it can not read any more data.
-   * This is in releation to the maxBufferSize vs the current readBuffer size.
+   * This is mainly to inform the SocketExecuter if the client should be allowed to select onReads.
    * 
    * @return true if more reads can be added, false if not.
    */
-  public boolean canRead() {
-    if(readBufferSize.get() > maxBufferSize) {
-      return false;
-    } 
-    return true;
-  }
+  public boolean canRead();
 
   /**
-   * Returns true if this client can accept more ByteBuffers to write.
-   * False is returned if the number of bytes pending write is greater than the max buffer size.
+   * Returns true if this client has data pending in its write buffers.  False if there is no data pending write.
+   *
+   * <p>Unlike canRead() and has no relation to if the write buffer is full or not.  Basically if there is anything in the 
+   * write buffer this should return true so the SocketExecuter knows to add this client to onWrite selector.</p>
    * 
-   * @return true if it can write more false if it cant.
+   * @return true if data is pending write, false if there is no data to write.
    */
-  public boolean canWrite() {
-    if(writeBufferSize.get() > 0) {
-      return true;
-    }
-    return false;
-  }
-  
+  public boolean canWrite();
```

Comment:
I agree they are unlikely to ever implement it.  But it's still something they will use.

That's the problem, because the way it is now, when ever they use it they see all these API's and need to understand what's relevant and what's not.

canWrite to me makes it sound like something I need to use, to know if I can apply additional write buffers.